sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/jwt/RestOperationsResourceRetriever.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.aad.implementation.jwt;
+
+import com.nimbusds.jose.util.Resource;
+import com.nimbusds.jose.util.ResourceRetriever;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.Assert;
+import org.springframework.web.client.RestOperations;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Arrays;
+
+/**
+ * A ResourceRetriever that accepts an RestOperations as a constructor parameter.
+ * Copied from org.springframework.security.oauth2.jwt.NimbusJwtDecoder.JwkSetUriJwtDecoderBuilder.RestOperationsResourceRetriever
+ */
+public class RestOperationsResourceRetriever implements ResourceRetriever {
+
+    private static final MediaType APPLICATION_JWK_SET_JSON = new MediaType("application", "jwk-set+json");
+
+    private final RestOperations restOperations;
+
+    public RestOperationsResourceRetriever(RestOperations restOperations) {
+        Assert.notNull(restOperations, "restOperations cannot be null");
+        this.restOperations = restOperations;
+    }
+
+    @Override
+    public Resource retrieveResource(URL url) throws IOException {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON, APPLICATION_JWK_SET_JSON));
+        ResponseEntity<String> response = getResponse(url, headers);
+        if (response.getStatusCodeValue() != 200) {
```

Comment:
Better to use `org.springframework.http.HttpStatus#OK` instead.

Reply:
Updated.